docs/resources/server/recipes/builtin.md

```diff
@@ -0,0 +1,368 @@
+# Built-In Recipe Types
+
+Minecraft provides a variety of recipe types and serializers out of the box for you to use. This article will explain each recipe type, as well as how to generate them.
+
+## Crafting
+
+Crafting recipes are typically made in crafting tables, crafters, or in modded crafting tables or machines. Their recipe type is `minecraft:crafting`.
+
+### Shaped Crafting
+
+Some of the most important recipes - such as the crafting table, sticks, or most tools - are created through shaped recipes. These recipes are defined by a crafting pattern or shape (hence "shaped") in which the items must be inserted. Let's have a look at what an example looks like:
+
+```json5
+{
+  "type": "minecraft:crafting_shaped",
+  "category": "equipment",
+  "pattern": [
+    "XXX",
+    " # ",
+    " # "
+  ],
+  "key": {
+    "#": {
+      "item": "minecraft:stick"
+    },
+    "X": {
+      "item": "minecraft:iron_ingot"
+    }
+  },
+  "result": {
+    "count": 1,
+    "id": "minecraft:iron_pickaxe"
+  }
+}
+```
+
+Let's digest this line for line:
+
+- `type`: This is the id of the shaped recipe serializer, `minecraft:crafting_shaped`.
+- `category`: This optional field defines the category in the crafting book.
+- `key` and `pattern`: Together, these define how the items must be put into the crafting grid.
+  - The pattern defines up to three lines of up to three-wide strings that define the shape. All lines must be the same length, i.e. the pattern must form a rectangular shape. Spaces can be used to denote slots that should stay empty.
+  - The key associates the characters used in the pattern with [ingredients][ingredient]. In the above example, all `X`s in the pattern must be iron ingots, and all `#`s must be sticks.
+- `result`: The result of the recipe. This is [an item stack's JSON representation][itemjson].
+- Not shown in the example is the `group` key. This optional string property creates a group in the recipe book. Recipes in the same group will be displayed as one in the recipe book.
+
+And then, let's have a look at how you'd generate this recipe:
+
+```java
+// We use a builder pattern, therefore no variable is created. Create a new builder by calling
+// ShapedRecipeBuilder#shaped with the recipe category (found in the RecipeCategory enum)
+// and a result item, a result item and count, or a result item stack.
+ShapedRecipeBuilder.shaped(RecipeCategory.EQUIPMENT, Items.IRON_PICKAXE)
```

Comment:
```suggestion
ShapedRecipeBuilder.shaped(RecipeCategory.TOOLS, Items.IRON_PICKAXE)
```
Needs to be reverted to `RecipeCategory.TOOLS`, I completely missed that vanilla for some stupid reason has two category enums for recipes and `ShapedRecipeBuilder` converts from `RecipeCategory` to `CraftingBookCategory` in `save()`. `equipment` in the JSON is the correct value for `RecipeCategory.TOOLS` in the builder